### PR TITLE
chore(aws-eks): add china regions support

### LINKED ISSUE #6481: Support EKS in China regions
According to the latest Amaozn EKS document [history](https://docs.aws.amazon.com/eks/latest/userguide/doc-history.html), it has been available in `cn-north-1` and `cn-northwest-1`. However, as `kubectl` layer is missing in SAR of `cn-north-1` we need some tweaks before we can successfully deploy it in China.

I will create a PR shortly to fix this issue.

### Error Log

```
 User: arn:aws-cn:iam::xxxxxxxxxxxx:user/pahud is not authorized to perform: serverlessrepo:CreateCloudFormationTemplate on resource: arn:aws:serverlessrepo:us-east-1:903779448426:applications/lambda-layer-kubectl
```

image

### Environment

  - **CLI Version      :**
  - **Framework Version:**
  - **OS               :**
  - **Language         :**

### Other

related PR from **eksctl**: https://github.com/weaveworks/eksctl/pull/1720

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by pahud**
- add `cn-north-1` and `cn-northwest-1` support
- `aws-lambda-layer-kubectl` bump to **2.0.0-beta2**

this PR fixes #6481 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by pahud** [comment]
copy @eladb 

I am not sure if it's appropriate to add the version bump into this PR. Please kindly help review it. Thanks.

File: packages/@aws-cdk/aws-eks/lib/kubectl-layer.ts [hunk 1]
```diff
@@ -75,4 +77,8 @@ export class KubectlLayer extends Construct implements lambda.ILayerVersion {
   public addPermission(_id: string, _permission: lambda.LayerVersionPermission): void {
     return;
   }
+
+  public isChina(): boolean {
+    return this.stack.region.startsWith('cn-');
```
Comment (eladb) [inline packages/@aws-cdk/aws-eks/lib/kubectl-layer.ts]: This can only be done if region is not a token.
```suggestion
    const region = this.stack.region;
    return !Token.isUnresolved(region) && region.startsWith('cn-');
```

File: packages/@aws-cdk/aws-eks/test/integ.eks-cluster.cn.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { Vpc } from '@aws-cdk/aws-ec2';
```
Comment (eladb) [inline packages/@aws-cdk/aws-eks/test/integ.eks-cluster.cn.ts]: I think it’s ok for this to just be tested through a unit test instead of an integ test. I am actually working on consolidating all the integ tests here into a couple of tests because it really hurts agility.

File: packages/@aws-cdk/aws-eks/test/integ.eks-cluster.defaults.expected.json [hunk 3]
```diff
@@ -1229,7 +1229,7 @@
               },
               "/",
               {
-                "Ref": "AssetParameters4dd58c778ce9bf0be6031537b4555ac9cbdd628b5770b5ec68016ff16b44428eS3Bucket3E4D83F8"
```
Comment (eladb) [inline packages/@aws-cdk/aws-eks/test/integ.eks-cluster.defaults.expected.json]: Why were these changed!? You haven’t updated the custom resource code?

File: packages/@aws-cdk/aws-eks/test/integ.eks-cluster.cn.ts (hunk 2 above)
Comment (pahud) [reply packages/@aws-cdk/aws-eks/test/integ.eks-cluster.cn.ts]: OK I will remove this integ test.

File: packages/@aws-cdk/aws-eks/test/integ.eks-cluster.defaults.expected.json (hunk 3 above)
Comment (pahud) [reply packages/@aws-cdk/aws-eks/test/integ.eks-cluster.defaults.expected.json]: Yes I have bumped the layer version from 2.0.0-beta1 to 2.0.0-beta2 in this PR so all  regions of classic partition as well as `aws-cn` will be using `2.0.0-beta2`.
https://github.com/aws-samples/aws-lambda-layer-kubectl/releases/tag/2.0.0-beta2

Is it the reason the references were changed? Should I do this in another PR?

https://github.com/aws/aws-cdk/blob/d202e796be8e309234e171d89aec8ec0ef294546/packages/%40aws-cdk/aws-eks/lib/kubectl-provider.ts#L37

Comment (eladb) [reply packages/@aws-cdk/aws-eks/test/integ.eks-cluster.defaults.expected.json]: OK got it. Yes, the template of the nested stack changed and therefore its artifact ID was invalidated.

I am okay with doing this in this PR. But... when you update your branch you'll notice that I've consolidated all integ tests into one, so your life is going to be easier (#6636).
